src/get-started/cryptocurrency-advanced.md

```diff
@@ -0,0 +1,835 @@
+            let history = currency_schema.wallet_history(&query.pub_key);
+            let proof = history.get_range_proof(0, history.len());
+```
+
+Next, we obtain transaction data for each history hash, transform them into the
+readable format and output them in the form of an array:
+
+```rust
+        let transactions: Vec<WalletTransactions> = history
+            .iter()
+            .map(|record| general_schema.transactions().get(&record).unwrap())
+            .map(|raw| WalletTransactions::tx_from_raw(raw).unwrap())
+            .collect::<Vec<_>>();
+
+        WalletHistory {
+            proof,
+            transactions,
+        }
+    });
+
+    Ok(WalletInfo {
+        block_proof,
+        wallet_proof,
+        wallet_history,
+    })
+}
+```
+
+We now have a complete proof for availability of a block in the blockchain, of a
+certain wallet in the database and
+said wallet’s history aggregated under the `WalletInfo` structure.
+
+The `post_transaction` defines the transactions processing logic, i.e. we
+convert a transaction into our internal
+Exonum-readable format and forward it into the network. The user in his turn
+receives back a hash of this transaction.
+
+```rust
+pub fn post_transaction(
+        state: &ServiceApiState,
+        query: WalletTransactions,
+    ) -> api::Result<TransactionResponse> {
+        let transaction: Box<dyn Transaction> = query.into();
+        let tx_hash = transaction.hash();
+        state.sender().send(transaction)?;
+        Ok(TransactionResponse { tx_hash })
+}
+```
+
+Finally, we `wire` function and call the above-mentioned methods with it:
+
+```rust
+pub fn wire(builder: &mut ServiceApiBuilder) {
+    builder
+        .public_scope()
+        .endpoint("v1/wallets/info", Self::wallet_info)
+        .endpoint_mut("v1/wallets/transaction", Self::post_transaction);
+}
+```
+
+## Interaction with the Service
+
+We have now described all the structural parts of our demo. The last step is to
+introduce the `main` function that will
+launch our service. It does this in the following manner:
+
+- imports our demo library described in `lib.rs` file together with
+  configuration service and `exonum` crate; and
+- creates nodes with two services – configuration service and our
+  cryptocurrency_advanced service:
+
+```rust
+extern crate exonum;
+extern crate exonum_configuration;
+extern crate exonum_cryptocurrency_advanced;
+
+use exonum::helpers::{self, fabric::NodeBuilder};
+use exonum_configuration as configuration;
+use exonum_cryptocurrency_advanced as cryptocurrency;
+
+fn main() {
+    exonum::crypto::init();
+    helpers::init_logger().unwrap();
+
+    let node = NodeBuilder::new()
+        .with_service(Box::new(configuration::ServiceFactory))
+        .with_service(Box::new(cryptocurrency::ServiceFactory));
+    node.run();
+}
+```
+
+As soon as you launch the demo with the `cargo run` command, the `NodeBuilder`
+will start to configure the network,
```

Comment:
I think it's worth describing what happens if we run the command several times (in parallel or in sequence): will a new network be built each time? Or will the blockchain continue from where it was stopped?